packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -536,17 +590,52 @@ class MemoryController extends DisposableController
     return libraryRoot;
   }
 
+  void addAllToNode(
+      Reference root, Map<String, List<HeapGraphClassLive>> allItems) {
+    allItems.forEach((libraryName, classes) {
+      LibraryReference libReference = root.children.singleWhere((library) {
+        return libraryName == library.name;
+      }, orElse: () => null);
+
+      // Library not found add to list of children.
+      libReference ??= LibraryReference(this, libraryName, classes);
+      root.addChild(libReference);
+
+      for (var actualClass in libReference.actualClasses) {
+        monitorClass(
+          className: actualClass.name,
+          message: 'computeAllLibraries',
+        );
+        final classRef = ClassReference(this, actualClass);
+        classRef.addChild(Reference.empty);
+
+        libReference.addChild(classRef);
+
+        // TODO(terry): Consider adding the ability to clear the table tree cache
+        // (root) to reset the level/depth values.
+        final classRefClassGroupBy = ClassReference(this, actualClass);
+        classRefClassGroupBy.addChild(Reference.empty);
+        classRoot.addChild(classRefClassGroupBy);
+      }
+    });
+  }
+
   Future getObject(String objectRef) async =>
       await serviceManager.service.getObject(
         _isolateId,
         objectRef,
       );
 
+  bool _gcing = false;
+  bool get isGcing => _gcing;
+
   Future<void> gc() async {
+    _gcing = true;
     await serviceManager.service.getAllocationProfile(
       _isolateId,
       gc: true,
     );
+    _gcing = false;
```

Comment:
put _gcing in a finally block so we never get stuck thinking we are gcing.

Reply:
Done.